Orange/widgets/widget.py

```diff
@@ -46,21 +48,10 @@ class WidgetMetaClass(type(QDialog)):
     #noinspection PyMethodParameters
     # pylint: disable=bad-classmethod-argument
     def __new__(mcs, name, bases, kwargs):
-        from Orange.canvas.registry.description import (
-            input_channel_from_args, output_channel_from_args)
-
         cls = super().__new__(mcs, name, bases, kwargs)
         if not cls.name: # not a widget
             return cls
 
-        cls.inputs = [input_channel_from_args(inp) for inp in cls.inputs]
-        cls.outputs = [output_channel_from_args(outp) for outp in cls.outputs]
```

Comment:
You're right. Also, this is preparation of the class, so it belongs to the metaclass, not to `get_widget_description`.